### PR TITLE
Adding viewport info for alexaRequests

### PR DESCRIPTION
**Pull request by freiSMS**
## Proposed changes
Added viewport interfaces for typescript + getAlexaDevice method, which gives back device description (according to alexaskill test page: small hub, large hub...)   

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

File: jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
-import {Input, JovoRequest, SessionData, Inputs, SessionConstants} from "jovo-core";
+import { Input, JovoRequest, SessionData, Inputs, SessionConstants } from "jovo-core";
 import _get = require('lodash.get');
 import _set = require('lodash.set');
+import * as util from 'util';
```
Comment (aswetlow) [inline jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts]: Only necessary with `inspect`. Can be removed.

File: jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
+import { Interface } from "readline";
```
Comment (aswetlow) [inline jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts]: Automatic import? Can be removed

File: jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts [hunk 3]
```diff
@@ -16,6 +18,7 @@ export interface Session {
 
 export interface Context {
     System: System;
+    Viewport: Viewport;
```
Comment (aswetlow) [inline jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts]: Do Echo Dot requests have the Viewport property?

File: jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts [hunk 4]
```diff
@@ -173,6 +212,62 @@ export class AlexaRequest implements JovoRequest {
 
     // JovoRequest implementation
 
+    getAlexaDevice(): string {
+        // console.log(alexaRequest);
```
Comment (aswetlow) [inline jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts]: Comments like these should be removed.

File: jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts [hunk 5]
```diff
@@ -173,6 +212,62 @@ export class AlexaRequest implements JovoRequest {
...
+        console.log("viewport : " + util.inspect(this.context!.Viewport));
+        let device = 'Echo - voice only';
+
+        if (this.context && this.context.Viewport ) {
+
+            device = "Unknow Device with Screen "+ this.context!.Viewport.pixelWidth + 'x' + this.context!.Viewport.pixelHeight;
+
+            if (this.context.Viewport.pixelWidth === 480 &&
+                this.context!.Viewport.pixelHeight === 480 &&
```
Comment (aswetlow) [inline jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts]: Bang operator is not needed here

File: jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts [hunk 6]
```diff
@@ -173,6 +212,62 @@ export class AlexaRequest implements JovoRequest {
...
+                this.context.Viewport.shape==='RECTANGLE') {
+                device = 'Alexa Large Hub' ;//'Echo Show 2nd gen';
+            }
+        }
+        console.log("Device in method = " + device);
+
+        return device;
+    }
+
+    getScreenResolution(alexaRequest: AlexaRequest): string {
```
Comment (aswetlow) [inline jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts]: Work in progress method?

File: jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts [hunk 7]
```diff
@@ -73,6 +76,42 @@ export interface System {
...
+    experiences: Experiences;
+    shape: string;
+    pixelWidth: number;
+    pixelHeight: number;
+    dpi: number;
+    currentPixelWidth: number;
+    currentPixelHeight: number;
+    touch: TouchMethods;
+    keyboard: InputMechanisms;
+    video: Codecs;
```
Comment (aswetlow) [inline jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts]: Optional?

File: jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts (hunk 3 above)
Comment (freiSMS) [reply jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts]: dots won't send it -> made optional

File: jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts [hunk 8]
```diff
@@ -73,6 +75,42 @@ export interface System {
     apiAccessToken?: string;
 }
 
+export interface Viewport {
+    experiences: Experiences;
+    shape: string;
```
Comment (aswetlow) [inline jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts]: shape: 'RECTANGLE' | 'ROUND';

File: jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts [hunk 9]
```diff
@@ -73,6 +75,42 @@ export interface System {
...
+    pixelWidth: number;
+    pixelHeight: number;
+    dpi: number;
+    currentPixelWidth: number;
+    currentPixelHeight: number;
+    touch?: TouchMethods;
```
Comment (aswetlow) [inline jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts]: `touch?: string[];`
or better
`type TouchMethod: 'SINGLE';`
`touch?: TouchMethod[];`

File: jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts [hunk 10]
```diff
@@ -73,6 +75,42 @@ export interface System {
...
+    keyboard?: InputMechanisms;
```
Comment (aswetlow) [inline jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts]: `keyboard?: string[];`
or
`type KeyboardMethod: 'DIRECTION';`
`keyboard?: KeyboardMethod[];`

File: jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts [hunk 11]
```diff
@@ -73,6 +75,42 @@ export interface System {
...
+    video?: Codecs;
```
Comment (aswetlow) [inline jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts]: `type VideoCodec: 'H_264_42' | 'H_264_41';`
`video?: {
   codecs: VideoCodec[];
}`

File: jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts [hunk 12]
```diff
@@ -173,6 +211,48 @@ export class AlexaRequest implements JovoRequest {
...
+                device = 'Alexa Medium Hub'; //'Echo Show 1st gen';
+            }
+
+            if (this.context.Viewport.pixelWidth === 1280 &&
+                this.context.Viewport.pixelHeight === 800 &&
+                this.context.Viewport.shape==='RECTANGLE') {
+                device = 'Alexa Large Hub' ;//'Echo Show 2nd gen';
+            }
+        }
+        console.log("Device in method = " + device);
```
Comment (aswetlow) [inline jovo-platforms/jovo-platform-alexa/src/core/AlexaRequest.ts]: Remove console.log()

**Pull request merged**